lib/modules/manager/pip-compile/extract.spec.ts

```diff
@@ -246,26 +246,35 @@ describe('modules/manager/pip-compile/extract', () => {
   });
 
   it('return sorted package files', async () => {
-    fs.readLocalFile.mockResolvedValueOnce(
-      getSimpleRequirementsFile('pip-compile --output-file=4.txt 3.in', [
-        'foo==1.0.1',
-      ]),
-    );
-    fs.readLocalFile.mockResolvedValueOnce('-r 2.txt\nfoo');
-    fs.readLocalFile.mockResolvedValueOnce(
-      getSimpleRequirementsFile('pip-compile --output-file=2.txt 1.in', [
-        'foo==1.0.1',
-      ]),
-    );
-    fs.readLocalFile.mockResolvedValueOnce('foo');
+    jest
+      .spyOn(fs, 'readLocalFile')
```

Comment:
why using spy here instead of simply using `fs.readLocalFile.mockImplementation` ?

Reply:
`readLocalFile` is overloaded, and `fs.readLocalFile.mockImplementation` trys to match the type to the first overload (the one that doesn't accept an encoding and returns a `Buffer`), which isn't the one we use.

Reply:
do this:
https://github.com/renovatebot/renovate/blob/25ae92d631d34507c194a2fd4353069bc03bb528/lib/modules/manager/bazel-module/bazelrc.spec.ts#L8